lib/src/picker_store.dart

```diff
@@ -33,67 +33,84 @@ class PickerStore extends flux.Store {
   }
 
   /// Returns the name of the member with the most recent pick from the active team.
-  String get lastPicker {
-    String lastMember = TEAM_MEMBER_NOT_FOUND;
-    int lastDate;
-    if (activeTeam == TEAM_NOT_FOUND) return lastMember;
-    members(_activeTeam).forEach((member) {
-      List<int> dates = pickDates(_activeTeam, member);
-      if (dates.isNotEmpty && (lastDate == null || dates.last > lastDate)) {
-        lastMember = member;
-        lastDate = dates.last;
+  GroupItem get lastPick {
+    int oldestDate = new DateTime.now().millisecondsSinceEpoch;
+    GroupItem lastItem;
+    getItemsInGroup(activeGroup).forEach((GroupItem item) {
+      if (item.lastPicked < oldestDate) {
+        oldestDate = item.lastPicked;
+        lastItem = item;
       }
     });
-    return lastMember;
+
+    return lastItem;
   }
 
   PickerStore(PickerActions actions) {
-    _teams = new List<String>();
-    _teamMembers = new Map<String, List<String>>();
-    _teamMemberPickDates = new Map<String, Map<String, List<int>>>();
-
-    actions
-        ..createTeam.listen(_createTeam)
-        ..removeTeam.listen(_removeTeam)
-        ..selectTeam.listen(_selectTeam)
-        ..createTeamMember.listen(_createTeamMember)
-        ..removeTeamMember.listen(_removeTeamMember)
-        ..selectTeamMember.listen(_selectTeamMember)
-        ..toggleTeamMemberActive.listen(_toggleTeamMemberActive)
-        ..pick.listen(_pick);
-  }
+    _groups = new List<Group>();
+    _groupItems = new List<GroupItem>();
 
-  // Action Handlers
+    fbClient = new FirebaseClient();
+    fbClient.signIn();
 
-  _createTeam(String team) {
+    // firebase subs
+    fbClient.fbGroups.onChildAdded.listen(groupAdded);
+    fbClient.fbGroupItems.onChildAdded.listen(groupItemAdded);
 
+    // action subs
+    actions
+      ..createGroup.listen(_createGroup)
+      ..removeGroup.listen(_removeGroup)
+      ..selectGroup.listen(_selectGroup)
+      ..createGroupItem.listen(_createGroupItem)
+      ..removeGroupItem.listen(_removeGroupItem)
+      ..selectGroupItem.listen(_selectGroupItem)
+      ..toggleGroupItemActive.listen(_toggleGroupItemActive)
+      ..pick.listen(_pick);
   }
 
-  _removeTeam(String team) {
-
+  // Firebase child listeners
+  void groupAdded(firebase.QueryEvent event) {
+    _groups.add(new Group.fromJson(event.snapshot.val()));
+    trigger();
   }
 
-  _selectTeam(String team) {
-
+  void groupItemAdded(firebase.QueryEvent event) {
+    _groupItems.add(new GroupItem.fromJson(event.snapshot.val()));
+    trigger();
   }
 
-  _createTeamMember(String member) {
+  // Action Handlers
 
+  _createGroup(String groupName) {
+    Group group = new Group(groupName);
+    _activeGroup = group;
```

Comment:
How does the timing for this work? Seems like you make a group active that isn't yet in _groups. So does the active Group data show up, and then later, once the firebase response comes back, it's added to _groups, and we see that list update?